process/baseProcessor.go

```diff
@@ -410,20 +411,35 @@ func (bp *BaseProcessor) isNodeOutOfSync(node *proxyData.NodeData) (bool, error)
 
 	nonce := nodeStatusResponse.Data.Metrics.Nonce
 	probableHighestNonce := nodeStatusResponse.Data.Metrics.ProbableHighestNonce
+	isReadyForVMQueries := isNodeReadyForVMQueries(nodeStatusResponse.Data.Metrics.AreVmQueriesReady)
 
 	probableHighestNonceLessThanOrEqualToNonce := probableHighestNonce <= nonce
 	nonceDifferenceBeyondThreshold := probableHighestNonce-nonce > nodeSyncedNonceDifferenceThreshold
 	isNodeOutOfSync := !probableHighestNonceLessThanOrEqualToNonce && nonceDifferenceBeyondThreshold
 
 	log.Info("node status",
 		"address", node.Address,
-		"shard", node.ShardId, "nonce",
-		nonce, "probable highest nonce",
-		probableHighestNonce, "is synced", !isNodeOutOfSync)
+		"shard", node.ShardId,
+		"nonce", nonce,
+		"probable highest nonce", probableHighestNonce,
+		"is synced", !isNodeOutOfSync,
+		"is ready for VM Queries", isReadyForVMQueries)
+
+	if !isReadyForVMQueries {
+		isNodeOutOfSync = true
+	}
 
 	return isNodeOutOfSync, nil
 }
 
+func isNodeReadyForVMQueries(metricValue string) bool {
```

Comment:
this is more like a sort of bool converter. Suggestion to rename it to `convertToBool(metricValue string) bool`